range_queries/mo_algorithm.cpp

```diff
@@ -0,0 +1,108 @@
+/**
+ *
+ * @file
+ * \brief A C++ Program to compute unique elements of ranges for different range
+ * queries
+ *
+ * \details
+ * The preprocessing part takes O(m Log m) time.
+ * Processing all queries takes O(n * √n) + O(m * √n) = O((m+n) * √n) time.
+ */
+#include <algorithm>
+#include <iostream>
+#include <cmath>
+#include <array>
+const int N = 1e6 + 5;
+std::array<int, N> a, bucket, cnt;
+// Variable to represent block size. This is made global so compare() of sort
+// can use it.
+int bucket_size;
+// Structure to represent a query range
+struct query {
+	int l, r, i;
+} q[N];
```

Comment:
```bash
/home/runner/work/C-Plus-Plus/C-Plus-Plus/range_queries/mo_algorithm.cpp:21:1: error: do not declare C-style arrays, use std::array<> instead [modernize-avoid-c-arrays,-warnings-as-errors].
```

Reply:
I used  std::array<int, N> q;
its throwing error.

Reply:
Try using `std::vector` instead.

Reply:
You're using `std::array` on line 16, but here you're using a C-style array.